src/sys/socket/sockopt.rs

```diff
@@ -887,6 +934,22 @@ mod test {
         assert!(a_cred.pid() != 0);
     }
 
+    #[cfg(any(target_os = "android", target_os = "linux"))]
+    #[test]
+    fn can_get_peersec_on_unix_socket() {
+        use super::super::*;
+
+        let (a, b) = socketpair(AddressFamily::Unix, SockType::Stream, None, SockFlag::empty()).unwrap();
+        let a_cred = getsockopt(a, super::PeerSecurity);
+        let b_cred = getsockopt(b, super::PeerSecurity);
+        match (a_cred, b_cred) {
+            (Ok(a_cred), Ok(b_cred)) => {
+                assert_eq!(a_cred, b_cred);
+            }
+            _ => {}
+        }
```

Comment:
This test would pass if both calls to `getsockopt` fail, is this desired behaviour?

Reply:
I believe my rationale was that we don't care if the underlying system doesn't support SO_PEERSEC, we just ignore the test in this case. Maybe a comment could clarify this behaviour.